forest/daemon/src/bundle/mod.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use forest_cli_shared::cli::Config;
+use forest_genesis::forest_load_car;
+use forest_networks::Height;
+use forest_shim::clock::ChainEpoch;
+use forest_utils::net::FetchProgress;
+use fvm_ipld_blockstore::Blockstore;
+use log::info;
+use tokio::{
+    fs::File,
+    io::{BufReader, BufWriter},
+};
+use tokio_util::compat::{FuturesAsyncReadCompatExt, TokioAsyncReadCompatExt};
+
+pub async fn load_bundles<DB>(epoch: ChainEpoch, config: &Config, db: DB) -> anyhow::Result<()>
+where
+    DB: Blockstore + Send + Sync + Clone + 'static,
+{
+    // collect bundles to load into the database.
+    let mut bundles = Vec::new();
+    if epoch < config.chain.epoch(Height::Hygge) {
+        bundles.push(get_actors_bundle(config, Height::Hygge).await?);
+    }
+
+    for bundle in bundles {
+        let result = forest_load_car(db.clone(), bundle.compat()).await?;
+        assert_eq!(
+            result.len(),
+            1,
+            "expected one root when loading actors bundle"
+        );
+        info!("Loaded actors bundle with CID: {}", result[0]);
+    }
+    Ok(())
+}
+
+/// Downloads the actors bundle (if not already downloaded) and returns a reader
+/// to it.
+// TODO Get it from IPFS instead of GitHub.
+pub async fn get_actors_bundle(config: &Config, height: Height) -> anyhow::Result<BufReader<File>> {
+    let bundle_info = config.chain.height_infos[height as usize]
+        .bundle
+        .as_ref()
+        .ok_or_else(|| anyhow::anyhow!("no bundle for epoch {}", config.chain.epoch(height)))?;
+
+    // This is the path where the actors bundle will be stored.
+    let bundle_path_dir = config
+        .client
+        .data_dir
+        .join("bundle")
```

Comment:
Maybe use **bundles** since we could have many of them. Just a suggestion though.

Is the `config.chain.name` really needed? For example we don't use it for **filecoin-proof-parameter**.

Reply:
Proof parameters are shared, and actor bundles are distinct in each network.